torch/csrc/utils/variadic.h

```diff
@@ -108,4 +108,50 @@ struct MakeIndices : MakeIndices<N-1, N-1, Is...> {};
 // 0 through N-1.
 template<size_t... Is>
 struct MakeIndices<0, Is...> { using indices = Indices<Is...>; };
+
+//===----------------------------------------------------------------------===//
+//                                 Utilities
+//===----------------------------------------------------------------------===//
+
+template <bool value, typename T = void>
+using enable_if_t = typename std::enable_if<value, T>::type;
+
+template <bool value, typename T = void>
+using disable_if_t = enable_if_t<!value, T>;
+
+template <typename T>
+using decay_t = typename std::decay<T>::type;
```

Comment:
Can you only define these if not already present, and if present do `using enable_if_t = std::enable_if_t` instead?